src/pages/SwapPage/BlockchainLog.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Card,
+  CardActions,
+  CardContent,
+  CardHeader,
+  Grid,
+  Typography
+} from "@material-ui/core";
+import React from "react";
+import { HtlcState, LedgerState } from "../../api/swapResource";
+import { Role } from "../../api/swapTypes";
+
+interface LedgerCardProps {
+  ledgerName: string;
+  state: LedgerState;
+  actions: Array<{ action: string; button: React.ReactNode }>;
+}
+
+function LedgerCard({ ledgerName, state, actions }: LedgerCardProps) {
+  let transactions = [
+    { verb: "deployed", hash: state.deploy_tx },
+    { verb: "funded", hash: state.fund_tx },
+    { verb: "redeemed", hash: state.redeem_tx },
+    { verb: "refunded", hash: state.refund_tx }
+  ];
+
+  if (state.deploy_tx === state.fund_tx) {
+    transactions = transactions.slice(1);
+  }
+
+  /* TODO: Link to a blockchain explorer if network is mainnet or
+     testnet */
+  /* TODO: Display transaction times if possible */
+  return (
+    <Card>
+      <CardHeader title={ledgerName} />
+      <CardContent>
+        {state.status !== HtlcState.NotDeployed ? (
+          transactions
+            .filter(tx => !!tx.hash)
+            .map(tx => {
+              return (
+                <React.Fragment key={tx.verb}>
+                  <Typography>HTLC {tx.verb} with transaction</Typography>
+                  <Typography noWrap={true}>{tx.hash}</Typography>
+                </React.Fragment>
+              );
+            })
+        ) : (
+          <Typography>No recorded events</Typography>
+        )}
+      </CardContent>
+      <CardActions>{actions.map(elem => elem.button)}</CardActions>
+    </Card>
+  );
+}
+
+interface BlockchainLogProps {
+  alphaState: LedgerState;
+  betaState: LedgerState;
+  role: Role;
+  actions: Array<{ action: string; button: React.ReactNode }>;
+}
+
+function BlockchainLog({
+  alphaState,
+  betaState,
+  role,
+  actions
+}: BlockchainLogProps) {
+  const alphaActions =
+    role === Role.Alice
+      ? actions.filter(
+          elem =>
+            elem.action === "deploy" ||
```

Comment:
:thinking: Not sure how good it is to include those here in terms of re-usability for future protocols. What was the reason why you are filtering them?

Reply:
> What was the reason why you are filtering them?

In this case I wanted actions to be displayed on the right column and I believe actions don't identify what ledger they're acting on so I had to look at the role instead.

> Not sure how good it is to include those here in terms of re-usability for future protocols. 

I agree with you and a lot of our components are already too specific to RFC003, but it's very hard to anticipate how other protocols would look like. This blockchain event log component is actually just a _RFC003_ blockchain event log. Maybe the name should reflect that.

Reply:
> This blockchain event log component is actually just a RFC003 blockchain event log.

Fair enough then, this one can have knowledge about those terms! :)